keps/sig-node/2625-cpumanager-policies-thread-placement/README.md

```diff
@@ -309,64 +273,146 @@ We expect no impact. The new policies are opt-in and separated by the existing o
-  - the number of pod ending up in Failed for SMTAlignmentError could be used to decide a rollback.
-* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?** Not Applicable.
-* **Is the rollout accompanied by any deprecations and/or removals of features,  APIs, fields of API types, flags, etc.?** No.
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Kubelet may fail to start. The kubelet may crash.
+
+###### What specific metrics should inform a rollback?
+
+We can use `cpu_manager_pinning_errors_total` to see all the allocation errors, irrespective of the specific reason though.
+In addition, we can use the logs: the number of pod ending up in Failed for SMTAlignmentError could be used to decide a rollback.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not Applicable.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
 
 ### Monitoring requirements
-* **How can an operator determine if the feature is in use by workloads?**
-  - Inspect the kubelet configuration of the nodes: check feature gates and usage of the new options
-* **What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?**
-  - No change
-* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?** N/A.
-* **Are there any missing metrics that would be useful to have to improve observability if this feature?** N/A.
 
+###### How can an operator determine if the feature is in use by workloads?
+
+- Check the metric `container_aligned_compute_resources_count` with the label `boundary=physical_cpu`
+- Inspect the kubelet configuration of the nodes: check feature gates and usage of the new options
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details:
+    - check metrics and their interplay:
+        * the metric `container_aligned_compute_resources_count` with the label `boundary=physical_cpu`
+        * the metric `cpu_manager_pinning_requests_total`
+        * the metric `cpu_manager_pinning_errors_total`
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - Metric name:
+    * the metric `container_aligned_compute_resources_count` with the label `boundary=physical_cpu`
+        * the metric `cpu_manager_pinning_requests_total`
+        * the metric `cpu_manager_pinning_errors_total`
+  - Components exposing the metric: kubelet
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+We can detail the pinning errors total with a new metric like `cpu_manager_errors_count` or
```

Comment:
For GA, I would expect that this metric is there already.

Reply:
Fair enough, we can add this as GA requirement.
Depends on: https://github.com/kubernetes/kubernetes/pull/129529

Reply:
So if it depends on that issue, then are we not promoting this to GA in this release? and adding a new metric as a new beta?

Reply:
While we can wait for another cycle, adding metrics in the context of graduation is something we did repeatedly in the past _and_ AFAIK is not controversial or uncommon.
I can also redo the work I did on 129529, it's just a bit wasteful and I'd prefer not to, but I can if this simplifies the flow.

Reply:
> While we can wait for another cycle, adding metrics in the context of graduation is something we did repeatedly in the past _and_ AFAIK is not controversial or uncommon. I can also redo the work I did on 129529, it's just a bit wasteful and I'd prefer not to, but I can if this simplifies the flow.

Actually no, we don't need to depend on this. On second look the redo is minimal. Will file a new PR. Thanks for the comment!

Reply:
Adding them in a separate PR.

Reply:
here: https://github.com/kubernetes/kubernetes/pull/129950